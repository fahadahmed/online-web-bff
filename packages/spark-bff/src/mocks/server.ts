import { setupServer } from 'msw/node';
import a2pCustomerHandlers from 'datasources/a2p/customer/mocks/handlers';
import accessHandlers from 'datasources/user/access/mocks/handlers';
import accountSummaryHandlers from 'datasources/customer/accountsService/mocks/handlers';
import addressHandlers from 'datasources/location/address/mocks/handlers';
import airpointsHandlers from 'datasources/airpoints/mocks/handlers';
import availableNumbers from 'datasources/shop/availableNumbers/mocks/handlers';
import balanceServiceHandlers from 'datasources/finance/balanceService/mocks/handlers';
import billPaymentHandlers from 'datasources/finance/paymentService/billPayment/mocks/handlers';
import cartServicesHandlers from 'datasources/shop/cart/mocks/handlers';
import checkoutServicesHandlers from 'datasources/shop/checkout/mocks/handlers';
import connectionDiagnosticsHandlers from 'datasources/line/assure/connectionDiagnostics/mocks/handlers';
import connectionPromiseHandler from 'datasources/line/assure/connectionPromise/mocks/handlers';
import contentServiceHandlers from 'datasources/utility/content/mocks/handlers';
import digitalBillHandlers from 'datasources/finance/digitalBill/mocks/handlers';
import identityServiceHandlers from 'datasources/user/identityService/mocks/handlers';
import lineConfiguarationHandlers from 'datasources/line/lineConfiguration/mocks/handlers';
import lineDetailServiceHandlers from 'datasources/line/lineDetails/mocks/handlers';
import lineDetailsHandlers from 'datasources/line/lineSummary/mocks/handlers';
import lineSpendHistoryHandlers from 'datasources/line/spend/mocks/handlers';
import lineSubscriptionsService from 'datasources/line/lineSubscriptionsService/mocks/handlers';
import lineUsageDetailsMocks from 'datasources/line/usage/usageDetailsService/mocks';
import lineUsageHistoryMocks from 'datasources/line/usage/usageHistoryService/mocks';
import lineUsageMocks from 'datasources/line/usage/mocks/handlers';
import lineUsageTransactionMocks from 'datasources/line/usage/usageTransactionsService/mocks';
import networkSettingsHandlers from 'datasources/technology/networkSettingsService/mocks/handlers';
import orderHandlers from 'datasources/shop/order/mocks/handlers';
import paymentServiceHandlers from 'datasources/finance/paymentService/mocks/handlers';
import pointOfAddressHandlers from 'datasources/location/pointOfInterest/mocks/handlers';
import pointOfOutageHandlers from 'datasources/location/pointOfInterest/outagesPointOfInterest/mocks/handlers';
import preferenceHandlers from 'datasources/user/preference/mocks/handlers';
import prepaidBalanceServiceHandlers from 'datasources/finance/prepaidBalanceService/mocks/handlers';
import productDetailServiceHandlers from 'datasources/product/details/mocks/handlers';
import productFiltersServiceHandlers from 'datasources/product/filters/mocks/handlers';
import productOffersHandlers from 'datasources/product/offers/mocks/handlers';
import productPromotionsMPDServiceHandlers from 'datasources/product/promotions/mpd/mocks/handlers';
import profileServiceHandlers from 'datasources/marketing/profileService/mocks/handlers';
import resourceHandlers from 'datasources/utility/resource/mocks/handlers';
import roamingRatesHandlers from 'datasources/product/roamingRates/mocks/handlers';
import subscriptionActivationHandlers from 'datasources/line/subscriptionActivationService/mocks/handlers';
import topupHandlers from 'datasources/finance/paymentService/topup/mocks/handlers';
import transactionsListHandlers from 'datasources/line/transactionsService/mocks/handlers';
import troubleshootHandlers from 'datasources/line/assure/connectionTroubleshooter/mocks/handlers';
import userInfoHandlers from 'datasources/user/userInfo/mocks/handlers';
import walletServiceHandlers from 'datasources/finance/walletService/mocks/handlers';

const handlers = [
  ...a2pCustomerHandlers,
  ...accessHandlers,
  ...accountSummaryHandlers,
  ...addressHandlers,
  ...availableNumbers,
  ...airpointsHandlers,
  ...balanceServiceHandlers,
  ...billPaymentHandlers,
  ...cartServicesHandlers,
  ...checkoutServicesHandlers,
  ...connectionDiagnosticsHandlers,
  ...connectionPromiseHandler,
  ...contentServiceHandlers,
  ...digitalBillHandlers,
  ...identityServiceHandlers,
  ...lineConfiguarationHandlers,
  ...lineDetailServiceHandlers,
  ...lineDetailsHandlers,
  ...lineSpendHistoryHandlers,
  ...lineSubscriptionsService,
  ...lineUsageDetailsMocks,
  ...lineUsageHistoryMocks,
  ...lineUsageMocks,
  ...lineUsageTransactionMocks,
  ...networkSettingsHandlers,
  ...orderHandlers,
  ...paymentServiceHandlers,
  ...pointOfAddressHandlers,
  ...pointOfOutageHandlers,
  ...preferenceHandlers,
  ...prepaidBalanceServiceHandlers,
  ...productDetailServiceHandlers,
  ...productFiltersServiceHandlers,
  ...productOffersHandlers,
  ...productPromotionsMPDServiceHandlers,
  ...profileServiceHandlers,
  ...resourceHandlers,
  ...roamingRatesHandlers,
  ...subscriptionActivationHandlers,
  ...topupHandlers,
  ...transactionsListHandlers,
  ...troubleshootHandlers,
  ...userInfoHandlers,
  ...walletServiceHandlers,
];

const server = setupServer(...handlers);
export default server;
