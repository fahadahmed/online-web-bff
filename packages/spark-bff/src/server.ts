import { ApolloServer } from 'apollo-server-fastify';
import {
  ApolloServerPluginLandingPageDisabled,
  ApolloServerPluginLandingPageGraphQLPlayground,
} from 'apollo-server-core';
import { logger } from 'apolloLogger';
import context from 'context';
import dataSources from 'datasources';
import { ApolloLoggerPlugin, ApplicationLogEvent } from 'logging';
import schema from 'schema';
import { formatResponse } from 'utils/formatResponse';
import { isDevelopmentEnvironment } from 'utils/isDevelopmentEnvironment';

process.on('uncaughtException', (err) => {
  logger.error(new ApplicationLogEvent({ message: err.message, context: {} }));
  throw err;
});

const apolloServer = new ApolloServer({
  schema,
  dataSources,
  formatResponse,
  logger,
  context,
  introspection: isDevelopmentEnvironment(),
  plugins: [
    new ApolloLoggerPlugin(
      `BFF Server Started. BUILD_ID: ${process.env.BUILD_BUILDID}`,
    ),
    isDevelopmentEnvironment()
      ? ApolloServerPluginLandingPageGraphQLPlayground()
      : ApolloServerPluginLandingPageDisabled(),
  ],
});

export default apolloServer;
