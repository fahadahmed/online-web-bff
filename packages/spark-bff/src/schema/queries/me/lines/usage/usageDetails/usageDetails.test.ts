import { createTestClient, gql } from 'test/apolloTestUtils';
import { server } from 'test/server';
import accessMocks from 'datasources/user/access/mocks/handlers';
import usageDetailsMocks from 'datasources/line/usage/usageDetailsService/mocks';
import lineSummaryMocks from 'datasources/line/lineSummary/mocks/handlers';

beforeEach(() => {
  server.use(...accessMocks, ...usageDetailsMocks, ...lineSummaryMocks);
});

test('returns all the Usage Details for line Number of line ', async () => {
  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query {
        me {
          line(lineNumber: "033227715") {
            lineNumber
            serviceType
            balanceManagement
            groupID
            groupProfile
            groupCapValue
            groupCapUnit
            usages {
              priority
              category
              productName
              billingStatus
              name
              uncapped
              types
              cap {
                unit
                value
              }
              used {
                unit
                value
                bySharer
              }
              remaining {
                unit
                value
              }
              expiry {
                date
                value
                unit
              }
              nextRenewalDate
              roaming
              rolloverData {
                value
                unit
              }
              rolloverExpiry {
                date
                value
                unit
              }
            }
          }
        }
      }
    `,
  });

  expect(res.errors).toBeUndefined();
  expect(res.data!.me.line).toMatchSnapshot();
});

test('returns value should be null for unKnown line number in lines ', async () => {
  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query {
        me {
          line(lineNumber: "000000") {
            lineNumber
            serviceType
            balanceManagement
            groupID
            groupProfile
            usages {
              priority
              category
              productName
              billingStatus
              name
              uncapped
              cap {
                unit
                value
              }
              used {
                unit
                value
                bySharer
              }
              remaining {
                unit
                value
              }
              expiry {
                date
                value
                unit
              }
              nextRenewalDate
              roaming
              rolloverData {
                value
                unit
              }
              rolloverExpiry {
                date
                value
                unit
              }
            }
          }
        }
      }
    `,
  });

  expect(res.data).toBeNull();
});
