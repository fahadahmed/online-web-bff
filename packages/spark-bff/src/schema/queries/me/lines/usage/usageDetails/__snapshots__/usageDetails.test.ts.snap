// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns all the Usage Details for line Number of line  1`] = `
Object {
  "balanceManagement": "PREPAID",
  "groupCapUnit": null,
  "groupCapValue": null,
  "groupID": "123",
  "groupProfile": null,
  "lineNumber": "033227715",
  "serviceType": "MOBILE",
  "usages": Array [
    Object {
      "billingStatus": "ACTIVE",
      "cap": Object {
        "unit": "MB",
        "value": 3584,
      },
      "category": "DATA",
      "expiry": Array [
        Object {
          "date": "Expires in 398 days",
          "unit": "MB",
          "value": 1745.67,
        },
        Object {
          "date": "Expires in 436 days",
          "unit": "MB",
          "value": 2000,
        },
      ],
      "name": "Rollover Data",
      "nextRenewalDate": "Renews in 155 days",
      "priority": 650,
      "productName": "Mobile $79.99 Rollover Plan",
      "remaining": Object {
        "unit": "MB",
        "value": 3584,
      },
      "roaming": false,
      "rolloverData": Object {
        "unit": "GB",
        "value": 3.657880859375,
      },
      "rolloverExpiry": Array [
        Object {
          "date": "2022-03-22T10:59:59Z",
          "unit": "MB",
          "value": 1745.67,
        },
        Object {
          "date": "2022-04-29T10:59:59Z",
          "unit": "MB",
          "value": 2000,
        },
      ],
      "types": Array [
        "ROLLOVER",
        "UNSHARED",
      ],
      "uncapped": false,
      "used": Object {
        "bySharer": null,
        "unit": "MB",
        "value": 0,
      },
    },
    Object {
      "billingStatus": null,
      "cap": Object {
        "unit": "MB",
        "value": 4500,
      },
      "category": "DATA",
      "expiry": Array [
        Object {
          "date": "Expires in 72 days",
          "unit": "GB",
          "value": 10,
        },
      ],
      "name": "AGGREGATED",
      "nextRenewalDate": null,
      "priority": null,
      "productName": null,
      "remaining": Object {
        "unit": "MB",
        "value": 1000,
      },
      "roaming": null,
      "rolloverData": Object {
        "unit": "GB",
        "value": 10,
      },
      "rolloverExpiry": Array [
        Object {
          "date": "2021-04-30T11:00:00.000Z",
          "unit": "GB",
          "value": 10,
        },
      ],
      "types": Array [
        "SOCIALISER",
      ],
      "uncapped": true,
      "used": Object {
        "bySharer": null,
        "unit": "MB",
        "value": 3500,
      },
    },
    Object {
      "billingStatus": null,
      "cap": Object {
        "unit": "MB",
        "value": 6852,
      },
      "category": "DATA",
      "expiry": Array [],
      "name": "AGGREGATED",
      "nextRenewalDate": null,
      "priority": null,
      "productName": null,
      "remaining": Object {
        "unit": "MB",
        "value": 1680.812322,
      },
      "roaming": false,
      "rolloverData": Object {
        "unit": "MB",
        "value": 0,
      },
      "rolloverExpiry": Array [],
      "types": Array [
        "AGGREGATED",
      ],
      "uncapped": false,
      "used": Object {
        "bySharer": null,
        "unit": "MB",
        "value": 5171.187678,
      },
    },
    Object {
      "billingStatus": null,
      "cap": Object {
        "unit": "MIN",
        "value": 700,
      },
      "category": "TALK",
      "expiry": Array [],
      "name": "AGGREGATED",
      "nextRenewalDate": null,
      "priority": null,
      "productName": null,
      "remaining": Object {
        "unit": "MIN",
        "value": 684,
      },
      "roaming": false,
      "rolloverData": Object {
        "unit": "MB",
        "value": 0,
      },
      "rolloverExpiry": Array [],
      "types": Array [
        "AGGREGATED",
      ],
      "uncapped": false,
      "used": Object {
        "bySharer": null,
        "unit": "MIN",
        "value": 16,
      },
    },
    Object {
      "billingStatus": null,
      "cap": null,
      "category": "TEXT",
      "expiry": Array [],
      "name": "AGGREGATED",
      "nextRenewalDate": null,
      "priority": null,
      "productName": null,
      "remaining": null,
      "roaming": false,
      "rolloverData": Object {
        "unit": "MB",
        "value": 0,
      },
      "rolloverExpiry": Array [],
      "types": Array [
        "AGGREGATED",
      ],
      "uncapped": true,
      "used": Object {
        "bySharer": null,
        "unit": "TEXT",
        "value": 39,
      },
    },
    Object {
      "billingStatus": "ACTIVE",
      "cap": null,
      "category": "DATA",
      "expiry": Array [
        Object {
          "date": "Expires in 199 days",
          "unit": "MB",
          "value": 1000,
        },
      ],
      "name": "Mobile Roaming Data Zone 6",
      "nextRenewalDate": null,
      "priority": 700,
      "productName": "Smart Prepaid",
      "remaining": null,
      "roaming": true,
      "rolloverData": Object {
        "unit": "MB",
        "value": 1000,
      },
      "rolloverExpiry": Array [
        Object {
          "date": "2021-09-04T11:59:59Z",
          "unit": "MB",
          "value": 1000,
        },
      ],
      "types": Array [
        "BASE_PLAN",
        "TIERED",
        "ROAMING",
        "UNSHARED",
      ],
      "uncapped": true,
      "used": Object {
        "bySharer": null,
        "unit": "MB",
        "value": 0,
      },
    },
    Object {
      "billingStatus": "ACTIVE",
      "cap": null,
      "category": "DATA",
      "expiry": Array [
        Object {
          "date": "Expires in 199 days",
          "unit": "MB",
          "value": 500,
        },
      ],
      "name": "Mobile Roaming Data Zone 6",
      "nextRenewalDate": null,
      "priority": 700,
      "productName": "Smart Prepaid",
      "remaining": null,
      "roaming": true,
      "rolloverData": Object {
        "unit": "MB",
        "value": 500,
      },
      "rolloverExpiry": Array [
        Object {
          "date": "2021-09-04T11:59:59Z",
          "unit": "MB",
          "value": 500,
        },
      ],
      "types": Array [
        "BASE_PLAN",
        "TIERED",
        "ROAMING",
        "UNSHARED",
      ],
      "uncapped": true,
      "used": Object {
        "bySharer": null,
        "unit": "MB",
        "value": 0,
      },
    },
    Object {
      "billingStatus": "ACTIVE",
      "cap": Object {
        "unit": "MIN",
        "value": 500,
      },
      "category": "TALK",
      "expiry": Array [
        Object {
          "date": "Expires in 199 days",
          "unit": "MIN",
          "value": 140,
        },
        Object {
          "date": "Expires in 226 days",
          "unit": "MIN",
          "value": 181,
        },
        Object {
          "date": "Expires in 255 days",
          "unit": "MIN",
          "value": 179,
        },
      ],
      "name": "Rollover Voice",
      "nextRenewalDate": null,
      "priority": 650,
      "productName": "Smart Prepaid",
      "remaining": Object {
        "unit": "MIN",
        "value": 500,
      },
      "roaming": false,
      "rolloverData": Object {
        "unit": "GB",
        "value": 500,
      },
      "rolloverExpiry": Array [
        Object {
          "date": "2021-09-04T11:59:59Z",
          "unit": "MIN",
          "value": 140,
        },
        Object {
          "date": "2021-10-01T10:59:59Z",
          "unit": "MIN",
          "value": 181,
        },
        Object {
          "date": "2021-10-30T10:59:59Z",
          "unit": "MIN",
          "value": 179,
        },
      ],
      "types": Array [
        "ROLLOVER",
        "NZ",
        "UNSHARED",
      ],
      "uncapped": false,
      "used": Object {
        "bySharer": null,
        "unit": "MIN",
        "value": 0,
      },
    },
    Object {
      "billingStatus": "ACTIVE",
      "cap": Object {
        "unit": "MIN",
        "value": 200,
      },
      "category": "TALK",
      "expiry": Array [],
      "name": "200 international minutes",
      "nextRenewalDate": "Renews in 147 days",
      "priority": 300,
      "productName": "$10 International Talk & Text Boost - 20 countries",
      "remaining": Object {
        "unit": "MIN",
        "value": 190,
      },
      "roaming": false,
      "rolloverData": Object {
        "unit": "MB",
        "value": 0,
      },
      "rolloverExpiry": Array [],
      "types": Array [
        "EXTRA",
        "INTERNATIONAL",
        "UNSHARED",
      ],
      "uncapped": false,
      "used": Object {
        "bySharer": null,
        "unit": "MIN",
        "value": 10,
      },
    },
    Object {
      "billingStatus": "ACTIVE",
      "cap": Object {
        "unit": "TEXT",
        "value": 100,
      },
      "category": "TEXT",
      "expiry": Array [],
      "name": "100 international texts",
      "nextRenewalDate": "Renews in 147 days",
      "priority": 300,
      "productName": "$10 International Talk & Text Boost - 20 countries",
      "remaining": Object {
        "unit": "TEXT",
        "value": 90,
      },
      "roaming": false,
      "rolloverData": Object {
        "unit": "MB",
        "value": 0,
      },
      "rolloverExpiry": Array [],
      "types": Array [
        "EXTRA",
        "INTERNATIONAL",
        "UNSHARED",
      ],
      "uncapped": false,
      "used": Object {
        "bySharer": null,
        "unit": "TEXT",
        "value": 10,
      },
    },
  ],
}
`;
